substrait/type/tests/TypeTest.cpp

```diff
@@ -87,34 +87,64 @@ TEST_F(TypeTest, decodeTest) {
         ASSERT_EQ(typePtr->signature(), "fchar<L1>");
       });
 
+  testDecode<FixedChar, false>(
+      "fixedchar<12>", [](const std::shared_ptr<const FixedChar>& typePtr) {
+        ASSERT_EQ(typePtr->length(), 12);
+        ASSERT_EQ(typePtr->signature(), "fchar<12>");
+      });
+
   testDecode<ParameterizedFixedBinary>(
       "fixedbinary<L1>",
       [](const std::shared_ptr<const ParameterizedFixedBinary>& typePtr) {
         ASSERT_EQ(typePtr->length()->value(), "L1");
         ASSERT_EQ(typePtr->signature(), "fbin<L1>");
       });
 
+  testDecode<FixedBinary, false>(
+      "fixedbinary<10>", [](const std::shared_ptr<const FixedBinary>& typePtr) {
+        ASSERT_EQ(typePtr->length(), 10);
+        ASSERT_EQ(typePtr->signature(), "fbin<10>");
+      });
+
   testDecode<ParameterizedVarchar>(
       "varchar<L1>",
       [](const std::shared_ptr<const ParameterizedVarchar>& typePtr) {
         ASSERT_EQ(typePtr->signature(), "vchar<L1>");
         ASSERT_EQ(typePtr->length()->value(), "L1");
       });
 
+  testDecode<Varchar,false>(
+      "varchar<5>", [](const std::shared_ptr<const Varchar>& typePtr) {
+        ASSERT_EQ(typePtr->signature(), "vchar<5>");
+        ASSERT_EQ(typePtr->length(), 5);
+      });
+
   testDecode<ParameterizedDecimal>(
       "decimal<P,S>",
       [](const std::shared_ptr<const ParameterizedDecimal>& typePtr) {
         ASSERT_EQ(typePtr->signature(), "dec<P,S>");
         ASSERT_EQ(typePtr->precision()->value(), "P");
         ASSERT_EQ(typePtr->scale()->value(), "S");
       });
+  testDecode<Decimal,false>(
```

Comment:
I assume something like:

```
const auto& type = ParameterizedType::decode("dec<P, S>", false)
```

should throw an exception (I think the stoi will throw).  Do we want to test that case?

Reply:
Added.